serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -102,7 +113,7 @@ def wait_for_request(self, timeout_seconds=None) -> bool:
             )
 
     def has_pending_requests(self) -> bool:
-        return self.queue or self.current_batch
+        return self.queue or self.current_batch or self.cancelled_requests
```

Comment:
This might be why cancelation was taking a while, we wont' actually step forward to cancel requests if we don't submit new work to the queue or the current batch.

Reply:
Makes sense.